apps/api/src/app/inbox/usecases/unsnooze-notification/unsnooze-notification.usecase.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  MessageRepository,
+  JobRepository,
+  JobStatusEnum,
+  ChannelTypeEnum,
+  JobEntity,
+  UserEntity,
+  OrganizationEntity,
+  EnvironmentEntity,
+} from '@novu/dal';
+import { Injectable, NotFoundException, InternalServerErrorException, NotImplementedException } from '@nestjs/common';
+import {
+  CreateExecutionDetails,
+  CreateExecutionDetailsCommand,
+  DetailEnum,
+  FeatureFlagsService,
+  PinoLogger,
+} from '@novu/application-generic';
+import { ExecutionDetailsSourceEnum, ExecutionDetailsStatusEnum, FeatureFlagsKeysEnum } from '@novu/shared';
+import { UnsnoozeNotificationCommand } from './unsnooze-notification.command';
+import { MarkNotificationAsCommand } from '../mark-notification-as/mark-notification-as.command';
+import { MarkNotificationAs } from '../mark-notification-as/mark-notification-as.usecase';
+import { InboxNotification } from '../../utils/types';
+
+const LOG_CONTEXT = 'UnsnoozeNotification';
+
+@Injectable()
+export class UnsnoozeNotification {
```

Comment:
The name and content of this usecase might seem similar to `process-unsnooze-job` usecase, where we actually unsnooze and "redeliver" the notification. 

The difference is that this usecase is more like "cancel", when user decides to "resolve" the snoozed notification manually. In such case the notification just returns to where it was, we dont need to update `createdAt` field or emit websocket even since the notification is not "new".

I'm wondering if we should use different naming.